### PR TITLE
Usability enhancements

### LINKED ISSUE #13: Allow input dictionary in Floris initialization
Currently, `Floris` expects a JSON input file path, but if someone wanted to, e.g., run FLORIS in batch mode, do parameter sweeps, etc., it may be useful to define the entire farm within a script. I suggest allowing the exact same parsing as the input file but on a Python dictionary. I'm sure it's already there under the hood since the JSON is read into a dictionary anyway.

### PR DESCRIPTION
**Pull request by petebachant**
I've made a bunch of changes to make FLORIS easier to use for my work. Resolves #13, supersedes #16 and #17.

### CODE REVIEW

**Comment by petebachant** [comment]
@jrannoni are you taking over maintenance on this package? It would be good to get this merged into `develop` then `master` before `master` moves too far ahead. Let me know if you have any suggestions for changes. These modifications have made it a bit easier for us to use FLORIS for our work.

**Comment by rafmudaf** [comment]
Hey @petebachant,

My apologies for the delay in getting this into FLORIS. We are planning to merge your changes, but I have a few things to add. Look out for a pull request to your fork and we can work together on getting to a good place with these improvements.

**Comment by petebachant** [comment]
Sounds good. Feel free to request changes from me here as well.

File: floris/farm.py [hunk 1]
```diff
@@ -59,29 +59,29 @@ class Farm():
...
-        # these attributes need special attention
-        self.wake_combination = WakeCombination(self.wake_combination)
-        self.wind_direction = np.radians(self.wind_direction - 270)
-
         turbine_dict = {}
         for c in list(zip(self.layout_x, self.layout_y)):
             turbine_dict[Coordinate(c[0], c[1])] = copy.deepcopy(turbine)
         self.turbine_map = TurbineMap(turbine_dict)
 
+        self.turbine = turbine
```
Comment (rafmudaf) [inline floris/farm.py]: why assign a particular turbine to the `farm` class?

Comment (petebachant) [reply floris/farm.py]: Good question! I need to see which commit put that in there.

**Comment by petebachant** [comment]
I think the notebooks will need to be updated for the longer module names.

**Comment by rafmudaf** [comment]
ah yes, I'll do that

**Comment by petebachant** [comment]
Alright, we may be good to go!

**Comment by rafmudaf** [comment]
Thanks for taking the time to do this @petebachant

**Pull request merged**